src/libpriv/rpmostree-rpm-util.c

```diff
@@ -882,41 +879,21 @@ rpmostree_get_refts_for_commit (OstreeRepo                *repo,
   return TRUE;
 }
 
-gboolean
-rpmostree_get_pkglist_for_root (int               dfd,
-                                const char       *path,
-                                RpmOstreeRefSack **out_refsack,
-                                GPtrArray        **out_pkglist,
-                                GCancellable     *cancellable,
-                                GError          **error)
-{
-  /* the DnfPackage objects don't have a ref on their DnfSack; make sure callers don't fall
-   * in that trap */
-  g_return_val_if_fail (out_pkglist == NULL || out_refsack != NULL, FALSE);
-
-  g_autoptr(RpmOstreeRefSack) refsack =
-    rpmostree_get_refsack_for_root (dfd, path, cancellable, error);
-  if (!refsack)
-    return FALSE;
-
-  hy_autoquery HyQuery query = hy_query_create (refsack->sack);
-  hy_query_filter (query, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
-  g_autoptr(GPtrArray) pkglist = hy_query_run (query);
-
-  if (out_refsack)
-    *out_refsack = g_steal_pointer (&refsack);
-  if (out_pkglist)
-    *out_pkglist = g_steal_pointer (&pkglist);
-  return TRUE;
-}
-
 static gint
 pkg_array_compare (DnfPackage **p_pkg1,
                    DnfPackage **p_pkg2)
 {
   return dnf_package_cmp (*p_pkg1, *p_pkg2);
 }
 
+GPtrArray*
+rpmostree_get_pkglist_for_refsack (RpmOstreeRefSack  *refsack)
+{
+  hy_autoquery HyQuery query = hy_query_create (refsack->sack);
+  hy_query_filter (query, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
+  return hy_query_run (query);
```

Comment:
Hmm but doesn't the pkglist need to hold a ref on the sack?

Reply:
Right, it doesn't have a ref on the sack, just like before. I added a precondition in #1134 to catch that, though I'm making it its own function here to make that more obvious. Hmm, but maybe it's not that obvious.

Tweaked the name and added a comment! ⬇️

Reply:
But don't the values in a `DnfPackage` point into the sack/pool?  AFAICS `DnfPackage` doesn't hold a ref on the pool though.  Hmm and there's `DnfPackageSet` which has-a map but not also a ref?

Reply:
They do, yeah. This was always a pitfall with this approach. This PR is just making that more obvious. Hmm, but actually, maybe we should just make it part of `RpmOstreeRefsack` instead. Let me see how that looks.

Reply:
I reduced these to `DnfSack` utility functions similar to what we already have in that same file. ⬇️
I think we're already comfortable with these functions all implying that their returned values will not outlive the `DnfSack` we pass in, right?